RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -270,7 +270,8 @@ void PulseShapeFitOOTPileupCorrection::setPUParams(bool   iPedestalConstraint, b
 }
 
 void PulseShapeFitOOTPileupCorrection::setPulseShapeTemplate(const HcalPulseShapes::Shape& ps, bool isHPD) {
-   if( cntsetPulseShape ) return;
+  // comment this otherwise he siPM is not correctly initialized
+  //  if( cntsetPulseShape ) return;
```

Comment:
May I suggest to edit this comment to clarify, smth like 
```
// initialize for every hit now to avoid incorrect settings for different channel types (HPD vs SiPM)
// FIXME: keep this as a reminder to improve and reduce CPU use
```